### DISCUSSION TITLE
CLIコマンドのドキュメント

### DISCUSSION THREAD

**Discussion by cocoide** [cocoide/commitify#18] [General]
suggest -s : コミットメッセージの生成
config -c: 設定オプションの表示
→ コミットメッセージのように複数選択の中から、OpenAIのToken設定ができるようにする
→Tokenを生成してください、みたいな対話型で入力できるとよさそう
→他にも、メッセージのStyleの設定、日本語or英語の設定なども可能にする
help -h: CLIの全体的な説明→cobraでうまく設定する
docs -d: コミットメッセージに関するドキュメント

**Comment by mochi-yu** [comment]
一般的なCLIツールだと、コマンドのエイリアスとしてハイフンを使うことは無いんじゃないかな？そもそもコマンドにはエイリアスは付けないイメージ。ハイフンを付けるものは、単語レベルのものはハイフン2個、頭文字だけだとハイフン1個とかが一般的かと思う（下記参照）

- `suggest`：コミットメッセージの生成
- `config`：各種の設定
  - 選択肢を出して設定項目を選んでもらう
    - 設定項目はAPIキー、日本語/英語、など
- `--help` `-h`：CLIのコマンドとオプション一覧を表示
- `--docs` `-d`：コミットメッセージに関するドキュメント
- `--version` `-v`：アプリのバージョン表示

**Reply by cocoide** [reply]
おっけい！色々調べてくれてありがとう
まだエイリアスはほぼ実装してないから、このままでいこう。

configの仕様はどうだと思う？
cofingの後、対話型にするか
それともコマンドを沢山作るか

**Reply by mochi-yu** [reply]
このままと言うのは、どれのことだろう？

**Reply by mochi-yu** [reply]
configはシンプルに設定項目選んで「入力してください」だけで良いのかなと自分は思うけど。
発表時間的にそこはあまり話せないんで、極論どちらでもって感じ。どちらでも良いならこだわらなくても良いのかなという想いが強い。

**Reply by cocoide** [reply]
> このままと言うのは、どれのことだろう？

今のブランチかな
suggestだけはエイリアスあるからそのコード消すだけでいいし、他はまだ未実装だからあんま気にする事ないって意味

**Reply by mochi-yu** [reply]
そうしたら結局、この先はどのコマンドで進めるってことになる？

**Reply by cocoide** [reply]
> configはシンプルに設定項目選んで「入力してください」だけで良いのかなと自分は思うけど。
> 
> 発表時間的にそこはあまり話せないんで、極論どちらでもって感じ。どちらでも良いならこだわらなくても良いのかなという想いが強い。

cofig系はまとめてやった方がコードの管理もしやすいかなって思った。発表はそうだけども
それ言ったらまあ色々意味ないよね

**Reply by cocoide** [reply]
> そうしたら結局、この先はどのコマンドで進めるってことになる？

suggestの機能を拡充していくのと
configでそれを成り立たせる
あとcommitifyって打った時の印象がぱっとしないから、色とか絵つけたりとか

コマンドを増やすっていうより今の機能をより良くしていく方針だね。もう少しアイデアとか出してもらえると助かる🙏

昨日discordでお願いしたのはどこまで進めてる感じ？

**Reply by mochi-yu** [reply]
これの命名にするのか、
```
suggest -s : コミットメッセージの生成
config -c: 設定オプションの表示
→ コミットメッセージのように複数選択の中から、OpenAIのToken設定ができるようにする
→Tokenを生成してください、みたいな対話型で入力できるとよさそう
→他にも、メッセージのStyleの設定、日本語or英語の設定なども可能にする
help -h: CLIの全体的な説明→cobraでうまく設定する
docs -d: コミットメッセージに関するドキュメント
```

こっちにするのか、
- `suggest`：コミットメッセージの生成
- `config`：各種の設定
  - 選択肢を出して設定項目を選んでもらう
    - 設定項目はAPIキー、日本語/英語、など
- `--help` `-h`：CLIのコマンドとオプション一覧を表示
- `--docs` `-d`：コミットメッセージに関するドキュメント
- `--version` `-v`：アプリのバージョン表示

それともまた変えたものにするのか、はどれだろう？

**Reply by cocoide** [reply]
> これの命名にするのか、
> 
> ```
> 
> suggest -s : コミットメッセージの生成
> 
> config -c: 設定オプションの表示
> 
> → コミットメッセージのように複数選択の中から、OpenAIのToken設定ができるようにする
> 
> →Tokenを生成してください、みたいな対話型で入力できるとよさそう
> 
> →他にも、メッセージのStyleの設定、日本語or英語の設定なども可能にする
> 
> help -h: CLIの全体的な説明→cobraでうまく設定する
> 
> docs -d: コミットメッセージに関するドキュメント
> 
> ```
> 
> 
> 
> こっちにするのか、
> 
> - `suggest`：コミットメッセージの生成
> 
> - `config`：各種の設定
> 
>   - 選択肢を出して設定項目を選んでもらう
> 
>     - 設定項目はAPIキー、日本語/英語、など
> 
> - `--help` `-h`：CLIのコマンドとオプション一覧を表示
> 
> - `--docs` `-d`：コミットメッセージに関するドキュメント
> 
> - `--version` `-v`：アプリのバージョン表示
> 
> 
> 
> それともまた変えたものにするのか、はどれだろう？

後者で全然いいよ

**Reply by mochi-yu** [reply]
おっけー

**Comment by mochi-yu** [comment]
参考までに、Gitコマンドのヘルプの例。`--version` `-v`みたいな情報表示するやつかオプションはハイフンとエイリアスがあって、普通に動作をするコマンドは単語だけ。

そういう意味では、設定も`config`だけでも良いかなとは思う。

```
usage: git [-v | --version] [-h | --help] [-C <path>] [-c <name>=<value>]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p | --paginate | -P | --no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           [--super-prefix=<path>] [--config-env=<name>=<envvar>]
           <command> [<args>]

These are common Git commands used in various situations:

start a working area (see also: git help tutorial)
   clone     Clone a repository into a new directory
   init      Create an empty Git repository or reinitialize an existing one

work on the current change (see also: git help everyday)
   add       Add file contents to the index
   mv        Move or rename a file, a directory, or a symlink
   restore   Restore working tree files
   rm        Remove files from the working tree and from the index

examine the history and state (see also: git help revisions)
   bisect    Use binary search to find the commit that introduced a bug
   diff      Show changes between commits, commit and working tree, etc
   grep      Print lines matching a pattern
   log       Show commit logs
   show      Show various types of objects
   status    Show the working tree status

grow, mark and tweak your common history
   branch    List, create, or delete branches
   commit    Record changes to the repository
   merge     Join two or more development histories together
   rebase    Reapply commits on top of another base tip
   reset     Reset current HEAD to the specified state
   switch    Switch branches
   tag       Create, list, delete or verify a tag object signed with GPG

collaborate (see also: git help workflows)
   fetch     Download objects and refs from another repository
   pull      Fetch from and integrate with another repository or a local branch
   push      Update remote refs along with associated objects
```

**Comment by mochi-yu** [comment]
CLI ツールのコマンド構文
https://docs.oracle.com/cd/E27813_01/html/E26290/gjuos.html

**Comment by mochi-yu** [comment]
issues #10
